### ISSUE TITLE
Cannot craft items on SMP

### ISSUE THREAD

**Issue by ghost** [LepkoQQ/EasyCrafting#13]
Easy Crafting is non-functional on SMP:

Using Forge Mod Loader version 4.5.65.500 on Minecraft 1.4.5.

Using Lepko-EasyCrafting 1.1.2.

The recipes are found, but when I click on the desired product, the cursor changes briefly (as if I could drag it) then reverts back to the standard "arrow cursor".

Edit: This is for custom items from other mods. I can craft vanilla minecraft items as expected.

**Comment by LepkoQQ** [comment]
can you be more specific? what mods? what items?

**Comment by ghost** [comment]
I have tested with ONLY Meteor Mod and then ONLY Buildcraft. So it is not a mod-specific issue. (ie: Meteor Mod and EasyCrafting were the only mods installed.)

EasyCrafting works for mod items when I am playing a single player game.

**Comment by ghost** [comment]
I am attempting to track it down via jSwat right now.

The Buildcraft item was Golden Conductive Pipe.

**Comment by ghost** [comment]
PacketHandlerServer.java:46 calls:

```
EasyRecipe recipe = RecipeHelper.getValidRecipe(result, ingredients);
```

recipe is null for items from other mods.

**Comment by ghost** [comment]
It appears that setAllRecipes() in RecipeHelper.java is executing too soon on the server. It is executing before all the mods are loaded, which causes to to miss recipes.

**Comment by LepkoQQ** [comment]
Thanks for looking in to it. I may have already fixed this in code on my end. Will double check before next release.

**Comment by LepkoQQ** [comment]
This was related to my mod loading before all other mods registered all of their recipes. Should be fixed in upcoming 1.5 version
